libraries/AP_RSSI/AP_RSSI.cpp

```diff
@@ -170,6 +170,18 @@ uint8_t AP_RSSI::read_receiver_rssi_uint8()
     return read_receiver_rssi() * 255; 
 }
 
+// Read the receiver RSSI value as an 8-bit integer with adhering to MAVLink standard
+// 0 represents weakest signal, 254 represents maximum signal, 255 represents invalid/unknown signal
+// we could rescale to range [0..254], but rather simply avoid 255, as this makes number consistent across telemetries
```

Comment:
Could you explain this reasoning in more detail, please?  The 0..254 thing seems compelling at first glance.

Reply:
well, if you prefer it scaled, no problem, I do not care
the reasoning here was that the value could e.g. also displayed by TLM and maybe other protocols, and so it would show the same for most parts

Reply:
> well, if you prefer it scaled, no problem, I do not care
> the reasoning here was that the value could e.g. also displayed by TLM and maybe other protocols, and so it would show the same for most parts

I'd think that this is just a transport value, and those showing it would translate back from the transport value into a normalised value - so 0-100% or 0-1.

Here's QGC doing just that: https://github.com/mavlink/qgroundcontrol/tree/master/src/FirmwarePlugin/APM/APMFirmwarePlugin.cc#L938  (incorrectly, past this patch ;-) )

... and MissionPlanner, just the same: https://github.com/ardupilot/MissionPlanner/blob/master/ExtLibs/ArduPilot/CurrentState.cs#L2812

Reply:
it may be the case they do it, but that's clearly a mistake
what they should do is to have a configurable scale, since it is explicitly NOT defined as to represent 0-100% !! Interpreting it this way is someone's own private definition put on top of it 
and which therefore causes all sorts of confusion
others may think it does or should represent db, and yet others may think whatever else

I really have problems with understanding why folks can't just go with what things are 
even the simplest things get tricky if two folks can't just agree that 2 is two for no sakes reason (and, frankly, that's why ap is such a mess then it comes to mavlink)

not all tel radios out there would ever produce 254 or even close to it, so they never could get even close to 100% in this interpretation ... if this makes sense to you, ok, it really doesn't to me, I want it to be close to 100% if I'm nearby and all is good
(you want an example? sik)

so, arguments-wise the situation isn't so easy ;)

but as said, if you prefer it scaled, no problem, I do not care ... it's a tiny difference at the end of the day
and not worth a long debate

what's really important is to exclude 255